lib/install/management/appliance.go

```diff
@@ -181,19 +181,39 @@ func (d *Dispatcher) deleteVM(vm *vm.VirtualMachine, force bool) error {
 		}
 	}
 
-	// Only retry VM destroy on CurrentAccess error
+	retryRemovalError := func(err error) bool {
+		return tasks.IsConcurrentAccessError(err) || tasks.IsInvalidPowerStateError(err)
+	}
+
+	// Only retry VM destroy on ConcurrentAccess error
```

Comment:
we might want to remove this comment (and maybe add some comments on why we're retrying on invalid power state I guess)